### PR TITLE
Add Codechecker commands

### LINKED ISSUE #86: CodeChecker integration
## Context

[CodeChecker](https://codechecker.readthedocs.io/en/latest/) is a static analysis infrastructure built on the LLVM/Clang Static Analyzer toolchain. It can scan your codebase with a three different tools (cppcheck, clangsa and clang-tidy) and produce report documents. This report document show what possible errors and mistakes are in your code.

Related issue that contains existing work: https://github.com/IRNAS/irnas-zephyr-template/issues/13

## Required steps / Implementation details

* Add CodeChecker dependencies (cppcheck, clang-tidy, clangsa) to the list of tools installed by the `east sys-setup` command.
* Implement workflows defined in the above link: developer usecase (analyze and diff) and CI use case.

## Definiton of Done

Above is done, reviewed and validated.

### PR DESCRIPTION
**Pull request by MarkoSagadin**
# Description

This PR adds new command to the East: `east codechecker` and its subcommands.

Users can now thoroughly analyze their code with the greatest open-source tools that exist on the market.

Currently we are still missing a support for `store` command for storing the results to the server.
This is planned.

## Areas of interest

Added code.

## Checklist

- [x] Source code documentation for all newly added or changed functions was added/updated.

## Related issues

Closes #86

## After-review steps

* Code author will merge the PR by himself/herself.

### CODE REVIEW

**Comment by MarkoSagadin** [comment]
Current dependencies on/for this PR:
* dev
  * **PR #87** <a href="https://app.graphite.dev/github/pr/IRNAS/irnas-east-software/87" target="_blank">Graphite</a> 
    * **PR #88** <a href="https://app.graphite.dev/github/pr/IRNAS/irnas-east-software/88" target="_blank">Graphite</a> 
      * **PR #89** <a href="https://app.graphite.dev/github/pr/IRNAS/irnas-east-software/89" target="_blank">Graphite</a>  👈

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/IRNAS/irnas-east-software/89?utm_source=stack-comment).

File: src/east/workspace_commands/codechecker_commands.py [hunk 1]
```diff
@@ -0,0 +1,202 @@
+# TODO:
+# codechecker commands:
+#   - check: runs analyze, performs filtering of the results and parses them.
+#       - Do not automatically do html report. if command is run without any args just
+#       show the results in the terminal.
+#       - To generate html report, run the command with --html-report and then manually
+#       open the html file.
```
Comment (TjazVracko) [inline src/east/workspace_commands/codechecker_commands.py:7]: Check is done, no?

File: src/east/workspace_commands/codechecker_commands.py [hunk 2]
```diff
@@ -0,0 +1,202 @@
...
+
+parse:
+  - --trim-path-prefix=/*/project
+  - --print-steps
+"""
+
+
+@click.command(**east_command_settings)
+@click.pass_obj
+def example_config(east):
```
Comment (TjazVracko) [inline src/east/workspace_commands/codechecker_commands.py:178]: You could add a `--create` flag here that would create the file and paste the example config into it.

File: src/east/workspace_commands/codechecker_helpers.py [hunk 3]
```diff
@@ -0,0 +1,178 @@
+import glob
+import os
+import plistlib as plist
+import re
+
+
+def check_for_compile_commands_json(east, compile_commands):
```
Comment (TjazVracko) [inline src/east/workspace_commands/codechecker_helpers.py:7]: Maybe check for the build dir first?

This way you could suggest building to the user 1st.
If the build dir exists and `compile_commands.json` does not, print the text you have here.

**Review by TjazVracko — APPROVED** [review]
I love it

File: src/east/workspace_commands/codechecker_commands.py (hunk 1 above)
Comment (MarkoSagadin) [reply src/east/workspace_commands/codechecker_commands.py:7]: Yep, I will delete that.

File: src/east/workspace_commands/codechecker_commands.py (hunk 2 above)
Comment (MarkoSagadin) [reply src/east/workspace_commands/codechecker_commands.py:178]: Good idea, will do that

File: src/east/workspace_commands/codechecker_helpers.py (hunk 3 above)
Comment (MarkoSagadin) [reply src/east/workspace_commands/codechecker_helpers.py:7]: Makes sense, will do that.

**Review by TjazVracko — APPROVED** [review]

**Pull request merged**